# Phase Rotation
### *Chapter 10 - Polyphase AC Circuits*

## Three-Phase Alternator

Let’s take the three-phase alternator design laid out earlier and watch what happens as the magnet rotates.

![three-phase-alternator](https://www.allaboutcircuits.com/uploads/articles/three-phase-alternator.jpg)

The phase angle shift of 120° is a function of the actual rotational angle shift of the three pairs of windings.

If the magnet is rotating clockwise, winding 3 will generate its peak instantaneous voltage exactly 120° (of alternator shaft rotation) after winding 2, which will hit its peak 120° after winding 1. The magnet passes by each pole pair at different positions in the rotational movement of the shaft.

Where we decide to place the windings will dictate the amount of phase shift between the windings’ AC voltage waveforms.

If we make winding 1 our “reference” voltage source for phase angle (0°), then winding 2 will have a phase angle of -120° (120° lagging, or 240° leading) and winding 3 an angle of -240° (or 120° leading).

## Phase Sequence

This sequence of phase shifts has a definite order. For clockwise rotation of the shaft, the order is 1-2-3 (winding 1 peak first, them winding 2, then winding 3). This order keeps repeating itself as long as we continue to rotate the alternator’s shaft.

![phase-sequence-plot](https://www.allaboutcircuits.com/uploads/articles/clockwise-rotation-phase.jpg)

*Note that the above notion*, and anything that depends on it, including any plot, assumes that the positions of $2$b and $2$a are swapped in the above diagram and that positions of winding pairs $2$ and $3$ are subsequently swapped.  Without this assumption, winding $2$ would have a phase angle of $-60^\circ$ (i.e. it would lag by $60^\circ$) and winding $3$ would have a phase angle of $-120^\circ$ (i.e. it would lag by $120^\circ$).

Below is the diagram with the right winding configuration.  

![correct-alternator](three-phase-alternator.svg)

Now, getting back to describing the phase sequence of the above depicted alternator, when we reverse direction -- that is to say, when the alternator is rotating in the counter-clockwise direction -- the associated phase sequence is reversed accordingly.

![reverse-phase-sequence](https://www.allaboutcircuits.com/uploads/articles/counterclockwise-rotation-phase.jpg)

## Phase Sequence Detector

Since voltmeters and ammeters would be useless in telling us what the phase rotation of an operating power system is, we need to have some other kind of instrument capable of doing the job.

One ingenious circuit design uses a capacitor to introduce a phase shift between voltage and current, which is then used to detect the sequence by way of comparison between the brightness of two indicator lamps in the figure below.

![phase-sequence-detector](https://www.allaboutcircuits.com/uploads/articles/phase-sequence-detector.jpg)

The two lamps are of equal filament resistance and wattage. The capacitor is sized to have approximately the same amount of reactance at system frequency as each lamp’s resistance.

If the capacitor were to be replaced by a resistor of equal value to the lamps’ resistance, the two lamps would glow at equal brightness, the circuit is balanced. However, the capacitor introduces a phase shift between voltage and current in the third leg of the circuit equal to 90°.

This phase shift, greater than 0° but less than 120°, skews the voltage and current values across the two lamps according to their phase shifts relative to phase 3.

In [1]:
import numpy as np

In [3]:
""" Define basic trig function that accept degrees by default. """

def sin(deg: float) -> float:
    return np.sin(np.radians(deg))

def cos(deg: float) -> float:
    return np.cos(np.radians(deg))

### With Resistor as Load \#3

Suppose all three loads have a resistance of $2650 \Omega$ and are connected to a power supply (an alternator) of $120\mathrm{V}$ with phase angles, from \#$1$ to \#$3$, each denoted as '$\theta_1$', '$\theta_2$', and '$\theta_3$' respectively, of $\theta_1 = 0$, $\theta_2 = 120^\circ$, and $\theta_3 = 240^\circ$.

### With Capacitor as Load \#3

When using **superposition theorem**, we evaluate the circuit one power supply at a time and then superimpose the results onto each other, keeping the polarities of each component whose values are being added in mind, to get the final result.

With **phase \#1** as the sole power supply, the circuit is of the form $120\mathrm{V}\angle\,0^\circ \text{--} R_1 \text{--}  C \text{//} R_2$.

The circuit equation is:  

$$ 120\mathrm{V}\angle\,0^\circ = I(R_1 + Z_2)$$

where the impedance of the parallel part of the circuit ($Z_2$) is given by:  

$$ Z_2 = \frac{1}{D_2}$$

and, 

\begin{align*}
    D_2 &= \frac{1}{Z_\mathrm{C}} + \frac{1}{R_2} = \frac{1}{Z_\mathrm{C}} + \frac{1}{R_1} \\
     &= \frac{1}{R_1}\,\angle\,90^\circ + \frac{1}{R_1} \\ \\
     &= \frac{1\angle\,90^\circ + 1}{R_1} \\ \\
     &= \frac{1 + j}{R_1}
\end{align*}     

In [2]:
np.degrees(np.arctan(1))

45.0

Thus, in polar form:  

$$ D_2 = \frac{\sqrt{2}}{R_1}\,\angle\,45^\circ$$

and,
$$ Z_2 = \frac{R_1}{\sqrt{2}}\,\angle\!-45^\circ$$

Via the above equation and susbtitution,  

\begin{align*}
    R_1 +Z_2 &= R_1\left(1 + \frac{1}{\sqrt{2}}\,\angle\!-45^\circ \right) \\ \\
     &= R_1(1.5 - 0.5j) \\
     &\approxeq 1.58R_1\,\angle\!-18.43^\circ
\end{align*}     

In [8]:
v1 = complex(1 + 1/np.sqrt(2) * cos(-45), 1/np.sqrt(2) * sin(-45))
mag1 = np.sqrt(v1.real**2 + v1.imag**2)
theta1 = np.degrees(np.arctan(v1.imag/v1.real))
(mag1, theta1)

(1.5811388300841898, -18.434948822922006)

Via the above approximate equality, the original circuit equation, and substitution,  

$$ 120\mathrm{V}\angle\,0^\circ \approxeq 1.58R_1I\,\angle\!-18.43^\circ$$

Which leads us to the approximate identity of $I_1$.  

$$ I \approxeq \frac{120\mathrm{V}}{1.58R_1}\,\angle\,18.43^\circ \approxeq 28.66\,\mathrm{mA}\,\angle\,18.43^\circ$$

In [9]:
120/(1.58*2650)

0.028660138524002867

The voltage drop across the first load, denoted as '$E_1$', is given by, via Ohm's Law, the above equation, and substitution,  

$$ E_1 = R_1I \approxeq \frac{120\mathrm{V}}{1.58}\,\angle\,18.43^\circ \approxeq 75.95\mathrm{V}\angle\,18.43^\circ$$

In [10]:
120/1.58

75.94936708860759

As for the other components that are in parallel with one another, via KVL, the above approximate equality, and substitution,  

$$ E_2 = E_3 = 120\mathrm{V}\angle\,0^\circ - E_1 \approxeq 120\mathrm{V}\angle\,0^\circ - 75.85\mathrm{V}\angle\,18.43^\circ$$

In [11]:
v2 = complex(120 - 75.85*cos(18.43), -75.85*sin(18.43))
v2

(48.040300373118484-23.979660748413163j)

Thus,
$$ E_2 = E_3 \approxeq (48.04 - 23.98j)\mathrm{V}$$

And, in polar form:  

$$ E_2 = E_3 \approxeq 53.69\mathrm{V}\angle\!-26.53^\circ$$

In [12]:
mag2 = np.sqrt(v2.real**2 + v2.imag**2)
theta2 = np.degrees(np.arctan(v2.imag/v2.real))
(mag2, theta2)

(53.692593432878944, -26.5264060243863)

With **phase \#2** as the sole power supply, the circuit is of the form $120\mathrm{V}\angle\,120^\circ \text{--} R_2\text{--}  R_1 \text{//} C$.

$$120\mathrm{V}\angle\,120^\circ \approxeq 1.58R_1I\,\angle\!-18.43^\circ$$

Thus, the current is given by:  

$$ I \approxeq \frac{120\mathrm{V}}{1.58R_1}\,\angle\,138.43^\circ$$

The voltage drop across the second load, denoted as '$E_2$', is given by, via Ohm's Law, the above equation, and substitution,  

$$ E_2 = R_1I \approxeq \frac{120\mathrm{V}}{1.58}\,\angle\,138.43^\circ \approxeq 75.95\mathrm{V}\angle\,138.43^\circ$$

As for the other components that are in parallel with one another, via KVL, the above approximate equality, and substitution,  

$$ E_1 = E_3 = 120\mathrm{V}\angle\,120^\circ - E_2 \approxeq 120\mathrm{V}\angle\,120^\circ - 75.85\mathrm{V}\angle\,138.43^\circ$$

In [13]:
v3 = complex(120*cos(120) - 75.85*cos(138.43), 120*sin(120) - 75.85*sin(138.43))
v3

(-3.2531548043008343+53.593950902762245j)

Thus, in rectangular form:  

$$ E_1 = E_3 \approxeq (-3.25 + 53.59j)\mathrm{V}$$

And, in polar form:  

$$ E_1 = E_3 \approxeq 53.69\mathrm{V}\angle\!-86.53^\circ$$

In [14]:
mag3 = np.sqrt(v3.real**2 + v3.imag**2)
theta3 = np.degrees(np.arctan(v3.imag/v3.real))
(mag3, theta3)

(53.69259343287895, -86.52640602438635)

With **phase \#3** as the sole power supply, the circuit is of the form $120\mathrm{V}\angle\,240^\circ \text{--} C \text{--}  R_1 \text{//} R_2$.

$$ 120\mathrm{V}\angle\,240^\circ = R_1I\big(1\angle\!-90^\circ + 0.5 \big) = R_1I(0.5 - j)$$

In polar form:  

$$120\mathrm{V}\angle\,240^\circ \approxeq 1.11R_1I\angle\!-63.43^\circ $$

In [15]:
mag4 = np.sqrt(0.5**2 + 1)
theta4 = np.degrees(np.arctan(-2))
(mag4, theta4)

(1.118033988749895, -63.43494882292201)

Which leaves us with a current of:  

$$ I \approxeq \frac{120\mathrm{V}}{1.11R_1}\,\angle\,303.43^\circ$$

Via the above equation, Ohm's Law, and substitution,  

$$ E_3 = Z_\mathrm{C}I \approxeq \frac{120\mathrm{V}}{1.11}\,\angle\,\big(303.43^\circ - 90^\circ \big)$$

In [16]:
(120/1.11, 303.43-90)

(108.1081081081081, 213.43)

Thus,
$$ E_3 \approxeq 108.19\mathrm{V}\angle\,213.43^\circ$$

The other three voltages are given by:  

$$ E_1 = E_2 = 120\mathrm{V}\angle\,240^\circ - E_3 \approxeq 120\mathrm{V}\angle\,240^\circ - 108.19\mathrm{V}\angle\,213.43^\circ$$

In [ ]:
v5 = complex()

$$ $$